### ISSUE TITLE
simplify DI.createInterface to be less error prone

### ISSUE THREAD

**Issue by xenoterracide** [aurelia/aurelia#976]
replace builder pattern
```
DI.createInterface('Foo').withDefault((b) -> b.cachedCallback(...));
DI.createInterface('Foo').noDefault();
DI.createInterface().withDefault((b) -> b.cachedCallback(...));
```
with a simple call
```ts
DI.createInterface('Foo', (b) -> b.cachedCallback(...));
DI.createInterface('Foo');
DI.createInterface((b) -> b.cachedCallback(...));
DI.createInterface();
```

it's shorter, less error prone if you forget noDefault, more intuitive. Only reason I could see a problem is if latter you decide it needs more parameters, but then it could easily take an object as a parameter.

discussed with @fkleuver

**Comment by xenoterracide** [comment]
lol, I love making duplicates #1040

**Comment by xenoterracide** [comment]
closed #1040
